design-proposals/enhancements.md

```diff
@@ -0,0 +1,103 @@
+# Overview
+
+Rework the existing KubeVirt design proposal process into a more
+formal - still lean - Enhancement Porposal process.
+
+## Motivation
+
+Today it's a [very small group](https://github.com/kubevirt/community/blob/main/OWNERS#L7-L14)
+which can approve design-proposals.
+While certain individuals can approve design proposals, once they are merged
+it is not clear who is owning them post-merge.
+
+In this proposal we are proposing to change the process and shape them
+around SIGs.
+At the heart this proposal is requiring
+
+1. designs to be sponsored by a SIG
+2. to make a SIG responsible for the design process (reviews of design, code, documentation)
+3. to make a SIG responsible for managing the design process (collaboration with other SIGs as needed)
+4. to make a SIG owning the feature once it has been merged. The SIG is responsible for maintaing, fixing, running, everything it.
+
+This has a few effects - see the following Goals section.
+
+## Goals
+
+1. The design process now has a mechanism to distribute designs among SIGs
+2. SIG approvers are empowered to approve designs, increase the approvers pool
+3. The ownership of an implemented feature becomes clear
+4. Ensure that designs converge (accept, reject)
+5. Formalize this process as an Virtualization Enhnacement Proposal (VEP) process
+
+## Non Goals
+
+1. Create unnecessary paperwork
+
+## Definition of Users
+
+* VEP Author - The person writing a design/enhancement proposal
+* SIG - A formal KubeVirt SIG
+* SIG Approver - A member of a SIG with approval permissions
+
+## User Stories
+
+* As a VEP Author I want to know who I can work with in order to move
+  my proposal forward
+* As a SIG I want to have a say in what is getting pushed into my domain
+  in order to make sure that we are able to maintain it
+* As a SIG Approver I want to ensure that a design is sound before a
+  VEP Author is approaching an implementation.
+
+## Repos
+
+- https://github.com/kubevirt/enhancements
+
+# Design
+
+Key elements:
+
+- Ownership: SIGs own a freature from it's inception (design) all the way to it's end
+- Approvals: SIG approvers will be allowed to approve designs
+- Responsibilities: SIG Approvers are responsible for driving a design, and connecitng it to other SIGs as needed
+
+Process elements:
+
+- VEP Author creates a GitHub Issue for getting a unique identifier and starting the process
+- VEP Author creates a PR to propose the design targeting a specific SIG
+- SIG decides on an approver to sheppered the VEP
```

Comment:
Later on the document talks about mirroring the OWNERS files from core `kubevirt/kubevirt` so I would assume the core approval group would end up with this responsibility.

It might be cleaner to define a meta SIG like `sig-architecture` in k8s who manage the KEP process and keep PRs flowing there?

Reply:
Generally I think each SIG needs to have a representative in order to make sure someone from the SIG is reviewing PRs/proposals/issues, otherwise everyone can say the next one will have a look. I'll have a look what we have written down for now and will propose this.

>It might be cleaner to define a meta SIG like sig-architecture in k8s who manage the KEP process and keep PRs flowing there?

I think this could be beneficial for more reasons, common solutions on common problems across SIGs and so on. Therefore +1 from me

Reply:
The SIG owns a PR. If the SIG is not responsible, then we should define a group which a VEP author can reach out to get some support.

I do _not_ think that all PRs should run through i.e. something like sig-architecture.
A core principle of this proposal is to distribute the load.

At best a group like a throectical sig-architecture could act as a moderator between SIG and VEP author, but must not be directly involved with a VEP, unless the VEP is owned by this (arch) SIG.

@davidvossel would it be good enough for you to say: "In case of an unresponsive SIG, the VPE author can consult the KubeVirt Maintainers in order to resolve the situation."?

Reply:
I've added:
> - KubeVirt Maintainers are responsible to support the owning SIG and VEP author in the case of conflicts and questions

Reply:
How about adding a label stating that a design proposal doesn't have an owning SIG yet? Something like `triage/needs-owning-sig`?

This way it would be more visible that the design proposal needs to be assigned + it would be easier querying such VEPs

Reply:
Yes, that's a good idea. How good are you in prow automation coding, Itamar?

Reply:
Thanks :)

FWIW - I think that we need approvers and "potential chairs" to really look into automation to simply help the process. the appovers and pr authors.